velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -1531,6 +1533,149 @@ TpchPlan TpchQueryBuilder::getQ19Plan() const {
+
+  const std::string receiptCommitFilter = "l_receiptdate > l_commitdate";
+
+  auto lineitemTwo =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(
+              kLineitem, lineitemTwoSelectedRowType, lineitemTwoFileColumns)
+          .capturePlanNodeId(lineitemTwoScanNodeId)
+          .project({"l_orderkey as l_orderkey_2", "l_suppkey as l_suppkey_2"})
+          .planNode();
+
+  auto lineitemThree =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(
+              kLineitem,
+              lineitemThreeSelectedRowType,
+              lineitemThreeFileColumns,
+              {},
+              receiptCommitFilter)
+          .capturePlanNodeId(lineitemThreeScanNodeId)
+          .project({"l_orderkey as l_orderkey_3", "l_suppkey as l_suppkey_3"})
+          .planNode();
+
+  auto nation = PlanBuilder(planNodeIdGenerator)
+                    .tableScan(
+                        kNation,
+                        nationSelectedRowType,
+                        nationFileColumns,
+                        {"n_name = 'SAUDI ARABIA'"})
+                    .capturePlanNodeId(nationScanNodeId)
+                    .planNode();
+
+  auto supplierJoinNation =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kSupplier, supplierSelectedRowType, supplierFileColumns)
+          .capturePlanNodeId(supplierScanNodeId)
+          .hashJoin(
+              {"s_nationkey"},
+              {"n_nationkey"},
+              nation,
+              "",
+              {"s_suppkey", "s_name", "s_nationkey"})
+          .planNode();
+
+  auto lineitemJoinSupplier =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(
+              kLineitem,
+              lineitemOneSelectedRowType,
+              lineitemOneFileColumns,
+              {},
+              receiptCommitFilter)
+          .capturePlanNodeId(lineitemOneScanNodeId)
+          .project({"l_orderkey as l_orderkey_1", "l_suppkey as l_suppkey_1"})
+          .hashJoin(
+              {"l_suppkey_1"},
+              {"s_suppkey"},
+              supplierJoinNation,
+              "",
+              {"l_orderkey_1",
+               "s_nationkey",
+               "l_orderkey_1",
+               "l_suppkey_1",
+               "s_name"})
+          .planNode();
+
+  auto plan = PlanBuilder(planNodeIdGenerator)
+                  .tableScan(
+                      kOrders,
+                      ordersSelectedRowType,
+                      ordersFileColumns,
+                      {"o_orderstatus = 'F'"})
+                  .capturePlanNodeId(ordersScanNodeId)
+                  .hashJoin(
+                      {"o_orderkey"},
+                      {"l_orderkey_1"},
+                      lineitemJoinSupplier,
+                      "",
+                      {"s_nationkey", "l_orderkey_1", "l_suppkey_1", "s_name"})
+                  .hashJoin(
+                      {"l_orderkey_1"},
+                      {"l_orderkey_2"},
+                      lineitemTwo,
```

Comment:
In both the semi and anti join the build sides are bigger than the probe side. But I guess we cannot avoid that because of the directionality of the join. Did you try with right semi join or other alternatives ?

Reply:
Thanks for the suggestion, at the time right semi join was not available so I had used left semi join. I have used the right semijoin now but the antijoin remains.